Zend/zend_compile.c

```diff
@@ -3025,7 +3068,15 @@ void zend_compile_assign(znode *result, zend_ast *ast) /* {{{ */
 				/* list($a, $b) = $a should evaluate the right $a first */
 				zend_compile_simple_var_no_cv(&expr_node, expr_ast, BP_VAR_R, 0);
 			} else {
-				zend_compile_expr(&expr_node, expr_ast);
+				if (zend_compile_list_assign_requires_w(var_ast)) {
+					if (zend_is_variable(expr_ast)) {
+						zend_compile_var(&expr_node, expr_ast, BP_VAR_W);
```

Comment:
This case also needs an extra `MAKE_REF` opcode. In this case it may make sense to only emit it if `expr_node.op_type != IS_CV`.

Reply:
Would you possibly have an example of code that would utilize this case, or rather, would expose why it'd be necessary?  So I can use it for understanding why / adding a test case.

Reply:
Working off the previous test-case, something like this should trigger it (not tested):

```
$ary = [[0, 1]];
[
    0 => &$a,
    ($ary["foo"] = 1) => &$b
] = $ary[0];
```

So basically the case where an array access has been shifted from the LHS to the RHS of the list assignment.